content/blog/2018/egress-mongo/index.md

```diff
@@ -0,0 +1,1120 @@
+    deployment "ratings-v2" created
+    {{< /text >}}
+
+1.  Route all the traffic destined to the _reviews_ service to its _v3_ version. You do this to ensure that the
+_reviews_ service always calls the _ratings_ service. In addition, route all the traffic destined to the _ratings_
+service to _ratings v2_ that uses your database.
+
+    Specify the routing for both services above by adding two
+    [virtual services](/docs/reference/config/istio.networking.v1alpha3/#VirtualService). These virtual services are
+    specified in `samples/bookinfo/networking/virtual-service-ratings-mongodb.yaml` of an Istio release archive.
+    ***Important:*** make sure you
+    [applied the default destination rules](/docs/examples/bookinfo/#apply-default-destination-rules) before running the
+     following command.
+
+    {{< text bash >}}
+    $ kubectl apply -f @samples/bookinfo/networking/virtual-service-ratings-db.yaml@
+    {{< /text >}}
+
+The updated architecture appears below. Note that the blue arrows inside the mesh mark the traffic configured according
+to the virtual services we added. According to the virtual services, the traffic is sent to _reviews v3_ and
+ _ratings v2_.
+
+{{< image width="80%" ratio="59.31%"
+    link="./bookinfo-ratings-v2-mongodb-external.svg"
+    caption="The Bookinfo application with ratings v2 and an external MongoDB database"
+    >}}
+
+Note that the MongoDB database is outside the Istio service mesh, or more precisely outside the Kubernetes cluster. The
+boundary of the service mesh is marked by a dashed line.
+
+### Access the webpage
+
+Access the webpage of the application, after
+[determining the ingress IP and port](/docs/examples/bookinfo/#determining-the-ingress-ip-and-port).
+
+Since you did not configure the egress traffic control yet, the access to the MongoDB service is blocked by Istio.
+This is why instead of the rating stars, the message _"Ratings service is currently unavailable"_ is currently
+ displayed below each review:
+
+{{< image width="80%" ratio="36.19%"
+    link="./errorFetchingBookRating.png"
+    caption="The Ratings service error messages"
+    >}}
+
+In the following sections you will configure egress access to the external MongoDB service, using different options for
+egress control in Istio.
+
+## Egress control for TCP
+
+Since [MongoDB Wire Protocol](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/) runs on top of TCP, you
+can control the egress traffic to your MongoDB as traffic to any other [external TCP service](/blog/2018/egress-tcp/). To
+control TCP traffic, a block of IPs in the [CIDR](https://tools.ietf.org/html/rfc2317) notation that includes the IP
+address of your MongoDB host must be specified. The caveat here is that the IP of the MongoDB host is not stable and
+could not be known in advance.
```

Comment:
As worded, this sentence is basically saying that this approach is not possible since MongoDB IPs are not stable or known. Is that what you meant to say? If not, it should say something like this instead:

```
The caveat here is that sometimes the IP of the MongoDB host is not stable or known in advance.
```

Reply:
Fixed in 2fb5405ee730e8dc7a5c004a4f8b05476bb6c49c.